src/util/TokenStreamBase.js

```diff
@@ -16,7 +16,7 @@ function TokenStreamBase(input, tokenData){
      * @property _reader
      * @private
      */
-    this._reader = input ? new StringReader(input.toString()) : null;
+    this._reader = input !== undefined ? new StringReader(input.toString()) : null;
```

Comment:
maybe its more safe to use `input != null` (http://stackoverflow.com/questions/2559318/how-to-check-for-an-undefined-or-null-variable-in-javascript#answer-15992131).